Integration with SQLStrings.jl

c42f opened the pull request:
This integrates the new https://github.com/JuliaComputing/SQLStrings.jl package (discussed in #213) into LibPQ, making it usable with the `execute()` function.

Currently based on #215 so if reviewing, just ignore the first commit.

This is quite a minimal integration. There's probably other options but this one is most straightforward if you don't mind the dependency.

c42f commented:
I'm not sure what's up with the docs build here, SQLStrings is registered at version 0.1.  That was quite recent though, maybe it's a stale registry somehow.

c42f commented:
Bump!

I'd like to get this merged, or alternatively get some feedback on a different more preferred approach.

quildtide commented:
@iamed2

iamed2 commented:
My apologies for completely missing this. I have started remonitoring this repo for activity. 

Happy to merge 7134a4ee383bf7195b649f137db6e6c64f217cf4, but it would be nice to also add a similar method for async_execute here: https://github.com/invenia/LibPQ.jl/blob/master/src/asyncresults.jl#L239

c42f commented:
Great, let's get this merged then.  I've rebased to remove the testing infrastructure from this PR and added `async_execute` along with a simple test of that.

I've tried to appease JuliaFormatter for the files in `src` but it also wants to reformat large sections of `runtests.jl` unrelated to this PR, so I'm not sure what to do about that.

iamed2 commented:
Yeah the JuliaFormatter review integration is broken and I don't like the actual correct suggestions anyway. This looks good to me!

iamed2 commented:
Ah current Invenia policy is to bump version numbers in merge requests, could you do that (to 1.12.0) now? Then I'll merge and registrate.

c42f commented:
Project.toml is bumped.
